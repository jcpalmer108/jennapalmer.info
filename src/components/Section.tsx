import { ReactNode } from 'react'
import { 
  Wrapper, 
  Header, 
  Line, 
  Label, 
  Title, 
  Children
} from "../styles/components/Section";
import line from "../assets/images/label-line.svg";

type SectionProps = {
  children: ReactNode,
  label?: string,
  light?: boolean,
  thin?: boolean,
  wide?: boolean,
  center?: boolean
}

export default function Section({ 
  label,
  children,
  light,
  thin,
  wide,
  center
}: SectionProps) {
  return (
    <Wrapper light={light} thin={thin} data-testid="Wrapper">
      {label && (
        <Header data-testid="Header" center={center} >
          <Label data-testid="Label">
            <Line src={line} alt="test"/>
            <Title>{label.toUpperCase()}</Title>
          </Label>
        </Header>
      )}
      <Children padded={!wide} data-testid="Children">{children}</Children>
    </Wrapper>
  )
}