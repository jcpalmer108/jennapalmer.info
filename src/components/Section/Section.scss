// MOBILE VIEW
.section {
  .section-inner {
    margin: auto;
    max-width: 1155px;
    transition: all 1s ease;

    .label {
      width: fit-content;
      padding-bottom: 10px;

      .label-inner {
        height: 12px;
        width: fit-content;
        column-gap: 10px;
        display: flex;
        align-items: center;
        justify-content: left;
        transition: all ease 1s;
        padding-bottom: 0px;

        img {
          width: 25px;
        }
      }
    }

    .centered {
      margin: auto;
    }

    .padded {
      padding: 20px;
      transition: all ease-in-out 1s;
    }
  }
}

.border {
  border-left: solid;
  border-right: solid;
  transition: all 1s ease-in-out;
}

.light {
  border-color: transparent;
  transition: all 1s ease-in-out;
}

.dark {
  border-color: transparent;
  transition: all 1s ease-in-out;
}

.thin {
  border-width: 0.5px;
  transition: all 1s ease-in-out;
}

.regular {
  border-width: 1px;
  transition: all 1s ease-in-out;
}

// DESKTOP VIEW
@media only screen and (min-width: 600px) {
  .section {
    .section-inner {
      .label {
        padding-bottom: 15px;

        .label-inner {
          font-size: 16px;
          height: 16px;
          transform: rotate(-90deg) translate(0px, 100%);
          transform-origin: bottom left;
          transition: all ease-in-out 1s;
          padding-bottom: calc(100% - 16px);
        }
      }

      .padded {
        padding: 30px 60px;
        transition: all ease-in-out 1s;
      }
    }
  }

  .border {
    border-left: solid;
    border-right: solid;
    transition: all 1s ease-in-out;
  }

  .light {
    border-color: transparent;
    transition: all 1s ease-in-out;
  }

  .dark {
    border-color: transparent;
    transition: all 1s ease-in-out;
  }

  .thin {
    border-width: 0.5px;
    transition: all 1s ease-in-out;
  }

  .regular {
    border-width: 1px;
    transition: all 1s ease-in-out;
  }
}

// 1155px and WIDER
@media only screen and (min-width: 1155px) {
  .border {
    border-left: solid;
    border-right: solid;
    transition: all 1s ease-in-out;
  }

  .light {
    border-color: white;
    transition: all 1s ease-in-out;
  }

  .dark {
    border-color: var(--silver-polish);
    transition: all 1s ease-in-out;
  }

  .thin {
    border-width: 0.5px;
    transition: all 1s ease-in-out;
  }

  .regular {
    border-width: 1px;
    transition: all 1s ease-in-out;
  }
}
