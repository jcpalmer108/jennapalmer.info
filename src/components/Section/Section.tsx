import React, { ReactNode } from "react";
// import { ReactComponent as LabelLine } from 'images/label-line.svg'

type SectionProps = {
  label?: string,
  className?: string,
  fullWidth?: boolean,
  light?: boolean,
  thin?: boolean,
  center?: boolean,
  children?: ReactNode
}

export default function Section({
  label, 
  className, 
  fullWidth, 
  light, 
  thin, 
  center, 
  children
}: SectionProps) {
  return (
    <div className={'section'.concat(className ? ` ${className}` : '')} data-testid="section">
      <div className={'section-inner border'.concat(
        ` ${light ? 'light' : 'dark'}`,
        ` ${thin ? 'thin' : 'regular'}`,
      )}>
        <div className={'content'.concat(fullWidth ? '' : ' padded')}>
          {
            label && (
              <div className={'label'.concat(center ? ' centered' : '')} data-testid="label">
                <div className="label-inner">
                  {/* <LabelLine /> */}
                  {label.toUpperCase()}
                </div>
              </div>
            )
          }
          {
            children && (
              <div className="children" data-testid="children">
                {children}
              </div>  
            )
          }
        </div>
      </div>
    </div>
  )
}