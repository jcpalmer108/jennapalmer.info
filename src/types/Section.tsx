export type WrapperProps = {
  light?: boolean,
  thin? : boolean,
  bottom?: boolean
}

export type HeaderProps = {
  center?: boolean,
}

export type ChildrenProps = {
  padded?: boolean
}
