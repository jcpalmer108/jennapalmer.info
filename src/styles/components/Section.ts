import styled from "styled-components";
import { desktopBreakpoint, contentWidth, silver_polish, tabletBreakpoint } from "../variables";
import { WrapperProps, HeaderProps, ChildrenProps } from "../../types/Section"

export const Wrapper = styled.div<WrapperProps>`
  max-width: ${contentWidth}px;
  margin: auto;
  border-left: solid;
  border-right: solid;
  border-color: transparent;
  border-width: 0px;

  @media (min-width: ${contentWidth}px) {
    border-color: ${props => props.light ? "white" : silver_polish};
    border-width: ${props => props.thin ? 0.5 : 1}px;
  }
`

export const Header = styled.div<HeaderProps>`
  width: fit-content;
  margin: ${props => props.center ? "auto" : "0px"};
  padding-top: 0px;

  @media (min-width: ${desktopBreakpoint}px) {
    margin: 0px;
    padding-top: 15px;
  }
`

export const Label = styled.div`
  height: 12px;
  font-size: 12;
  width: fit-content;
  column-gap: 10px;
  display: flex;
  align-items: center;
  justify-content: left;
  padding-left: 20px;
  padding-bottom: 0px;
  padding-top: 50px;

  @media (min-width: ${tabletBreakpoint}px) {
    height: 14px;
    font-size: 14
  }

  @media (min-width: ${desktopBreakpoint}px) {
    height: 16px;
    font-size: 16;
    transform: rotate(-90deg) translate(0px, 100%);
    transform-origin: bottom left;
    padding-bottom: calc(100% - 16px);
    padding-left: 0px;
  }

`

export const Line = styled.img`
  width: 25px;
`
export const Title = styled.span`
  font-size: 12px;

  @media (min-width: ${tabletBreakpoint}px) {
    font-size: 14px;
  }

  @media (min-width: ${desktopBreakpoint}px) {
    font-size: 16px;
  }

`

export const Children = styled.div<ChildrenProps>`
  padding: ${props => props.padded ? 20 : 0}px;
`
