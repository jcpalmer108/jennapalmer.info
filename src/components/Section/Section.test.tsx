import React from 'react';
import { render, screen } from '@testing-library/react';
import Section from './Section';

test('when given params it should render as expected', () => {
  render(
    <Section label="test" className="test-class" fullWidth light thin center>
      <p>test child</p>
      <p>test child</p>
    </Section>
  );
  expect(screen.getByTestId('section')).toMatchSnapshot();
});

test('when not given params it should render as expected', () => {
  render(<Section />);
  expect(screen.getByTestId('section')).toMatchSnapshot();
});
