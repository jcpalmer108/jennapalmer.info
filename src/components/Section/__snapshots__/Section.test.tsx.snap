// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when given params it should render as expected 1`] = `
<div
  class="section test-class"
  data-testid="section"
>
  <div
    class="section-inner border light thin"
  >
    <div
      class="content"
    >
      <div
        class="label centered"
        data-testid="label"
      >
        <div
          class="label-inner"
        >
          TEST
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`when not given params it should render as expected 1`] = `
<div
  class="section"
  data-testid="section"
>
  <div
    class="section-inner border dark regular"
  >
    <div
      class="content padded"
    />
  </div>
</div>
`;
